newdle/client/src/components/creation/EditPage.js

```diff
@@ -0,0 +1,46 @@
+import React, {useEffect} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import {Redirect, Route, Switch, useParams} from 'react-router-dom';
+import {getCreatedNewdle, isLoggedIn} from '../../selectors';
+import {abortCreation, fetchNewdle} from '../../actions';
+import {usePageTitle} from '../../util/hooks';
+import {Loader} from 'semantic-ui-react';
+import ParticipantsStep from './ParticipantsStep';
+import TimeslotsStep from './timeslots';
+import FinalStep from './FinalStep';
+
+export default function EditPage() {
+  const {code: newdleCode} = useParams();
+  const isUserLoggedIn = useSelector(isLoggedIn);
+  const newdle = useSelector(getCreatedNewdle);
+  const dispatch = useDispatch();
+  usePageTitle('Editing newdle');
+
+  useEffect(() => {
+    if (newdleCode) {
+      // TODO: We should re-use the existing newdle data instead. However, abortCreation will clear the state,
+      //  causing an edit -> summary -> edit to become empty if we remove the line below
+      //  Ideally, we should move abortCreation() - state flushing - to happen on mount instead of unmount.
+      dispatch(fetchNewdle(newdleCode, true));
+    }
+
+    return () => {
+      dispatch(abortCreation());
+    };
+  }, [dispatch, newdleCode]);
+
+  if (!isUserLoggedIn) {
+    return <Redirect to="/" />;
+  }
+  if (!newdle) {
+    return <Loader active />;
+  }
+
+  return (
+    <Switch>
+      <Route exact path="/newdle/:code/edit" render={() => <TimeslotsStep isEditing />} />
```

Comment:
maybe it'd be cleaner to have `.../timeslots` there?

Reply:
Yes, but seems to weird to have `edit` pointing nowhere. Perhaps defaulting to `timeslots`?

Reply:
yeah, you could redirect `.../edit` to `/edit/timeslots` (i guess that's what you meant with defaulting to timeslots?)

Reply:
Exactly.